apps/web-giddh/src/app/shared/header/header.component.ts

```diff
@@ -1989,7 +1989,9 @@ export class HeaderComponent implements OnInit, AfterViewInit, OnDestroy, AfterV
      * @memberof HeaderComponent
      */
     public navigateToPreviousRoute(): void {
-        this.location.back();
+        if (document.referrer) {
```

Comment:
If there is no referrer, we should redirect to dashboard just like we did for settings sidebar back button. Please confirm.

Reply:
Done